catalyst-gateway/bin/src/db/index/queries/staked_ada/get_txo_by_stake_address.rs

```diff
@@ -69,12 +82,64 @@ impl GetTxoByStakeAddressQuery {
     /// Executes a get txo by stake address query.
     pub(crate) async fn execute(
         session: &CassandraSession, params: GetTxoByStakeAddressQueryParams,
-    ) -> anyhow::Result<TypedRowStream<GetTxoByStakeAddressQuery>> {
-        let iter = session
-            .execute_iter(PreparedSelectQuery::TxoByStakeAddress, params)
+    ) -> anyhow::Result<Vec<GetTxoByStakeAddressQuery>> {
+        if session.is_persistent() {
+            if let Some(rows) = ASSETS_CACHE.get(&params.stake_address).await {
+                return Ok(rows);
+            }
+        }
+
+        let rows: Vec<_> = session
+            .execute_iter(PreparedSelectQuery::TxoByStakeAddress, &params)
             .await?
-            .rows_stream::<GetTxoByStakeAddressQuery>()?;
+            .rows_stream::<GetTxoByStakeAddressQuery>()?
+            .map_err(Into::<anyhow::Error>::into)
+            .try_collect()
+            .await?;
+
+        // update cache
+        if session.is_persistent() {
+            ASSETS_CACHE
+                .insert(params.stake_address, rows.clone())
+                .await;
+        }
+
+        Ok(rows)
+    }
+
+    /// Update spent UTXO data. Returns true if entry was found and updated.
```

Comment:
Either the return value or the comment should be updated, because currently the function returns `()`.